### PR TITLE
Add a gh action for aws deployment

### PR DESCRIPTION
**Pull request by gmoynihan88**
Description
This enables GitHub actions to publish to AWS, production and staging 

How to review
...

Testing
...

### CODE REVIEW

File: .github/workflows/aws-deployment.yml [hunk 1]
```diff
@@ -0,0 +1,95 @@
+---
+    name: AWS Deployment
```
Comment (anl) [inline .github/workflows/aws-deployment.yml:2]: Indentation seems off... this should be left-justified, and two spaces per indentation.

File: .github/workflows/aws-deployment.yml [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+                description: "Environment to deploy to"
+                default: "staging"
+                options:
+                    - staging
+                    - production
+                required: true
+                type: choice
+
+    env:
+      AWS_ACCOUNT_ID: ${{ vars.AWS_PUBLIC_DATA_RELEASES_ACCOUNT_ID }}
```
Comment (anl) [inline .github/workflows/aws-deployment.yml:17]: Double-checking - are these variables set?

File: .github/workflows/aws-deployment.yml [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+      AWS_REGION: ${{ vars.AWS_DEFAULT_REGION }}
+      STAGING_CLOUDFRONT_DISTRIBUTION_ID: ${{ secrets.STAGING_CLOUDFRONT_DISTRIBUTION_ID }}
+      STAGING_S3_BUCKET: s3://staging.timelapse.allencell.org
```
Comment (anl) [inline .github/workflows/aws-deployment.yml:20]:
```suggestion
      STAGING_S3_BUCKET: s3://staging.biofile-finder.allencell.org
```

File: .github/workflows/aws-deployment.yml [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+      PRODUCTION_CLOUDFRONT_DISTRIBUTION_ID: ${{ secrets.PRODUCTION_CLOUDFRONT_DISTRIBUTION_ID }}
+      PRODUCTION_S3_BUCKET: s3://timelapse.allencell.org
```
Comment (anl) [inline .github/workflows/aws-deployment.yml:22]:
```suggestion
      PRODUCTION_S3_BUCKET: s3://biofile-finder.allencell.org
```

File: .github/workflows/aws-deployment.yml [hunk 5]
```diff
@@ -0,0 +1,95 @@
...
+
+    jobs:
+        build:
+          runs-on: ubuntu-latest
+
+          steps:
+            - name: Checkout
+              uses: actions/checkout@f43a0e5ff2bd294095638e18286ca9a3d1956744
+
+            - uses: actions/setup-node@60edb5dd545a775178f52524783378180af0d1f8
```
Comment (anl) [inline .github/workflows/aws-deployment.yml:36]: @SeanLeRoy These build steps will presumably need adjusting for your workflow.

**Review by anl — CHANGES_REQUESTED** [review]
Looks like there's some copy-paste issues and some indentation fixes needed.  @SeanLeRoy should weigh in on how the actual _build_ is done, those steps will likely need to be adjusted to match.

File: .github/workflows/aws-deployment.yml (hunk 3 above)
Comment (gmoynihan88) [reply .github/workflows/aws-deployment.yml:20]: Done, thanks!

File: .github/workflows/aws-deployment.yml (hunk 5 above)
Comment (SeanLeRoy) [reply .github/workflows/aws-deployment.yml:36]: Ah yeah this seems a bit different, I'll likely need to workshop this a bit. Does it just need to produce some build artifact at `./dist`?

File: .github/workflows/aws-deployment.yml (hunk 1 above)
Comment (gmoynihan88) [reply .github/workflows/aws-deployment.yml:2]: There was some double-indenting on the source I began with.  I'm going to fix it in vs code 

https://raw.githubusercontent.com/allen-cell-animated/nucmorph-colorizer/main/.github/workflows/aws-deploy.yml

File: .github/workflows/aws-deployment.yml (hunk 5 above)
Comment (anl) [reply .github/workflows/aws-deployment.yml]: Yes - the deploy step picks them up from there (we could also change paths, if need be).

File: .github/workflows/aws-deployment.yml (hunk 2 above)
Comment (anl) [reply .github/workflows/aws-deployment.yml]: Looks like these would need to go in Repository Variables, here: https://github.com/AllenInstitute/aics-fms-file-explorer-app/settings/variables/actions

**Review by anl — APPROVED** [review]

File: .github/workflows/aws-deployment.yml (hunk 5 above)
Comment (SeanLeRoy) [reply .github/workflows/aws-deployment.yml]: How should I go about testing this - if there is nothing in the `./dist` will the build fail (as in, could I safely run this workflow to see if I am placing the files in the correct way and place)?

File: .github/workflows/aws-deployment.yml (hunk 2 above)
Comment (gmoynihan88) [reply .github/workflows/aws-deployment.yml]: Looks good to me

File: .github/workflows/aws-deployment.yml (hunk 5 above)
Comment (SeanLeRoy) [reply .github/workflows/aws-deployment.yml]: https://github.com/AllenInstitute/aics-fms-file-explorer-app/pull/113

**Review by SeanLeRoy — APPROVED** [review]

**Pull request merged**